File: exemplos/js/10_erro_problem_details.mjs
// Formato de erro da API: RFC 9457 (application/problem+json) com os campos da casa junto
import { chamar, exigir } from "./_apoio.mjs";

const { status, cabecalhos, objeto } = await chamar("/api/eu", { chave: "plat_" + "z".repeat(43) });
exigir(status === 401, `esperava 401 com chave inválida, veio ${status}: ${JSON.stringify(objeto)}`);
const tipo = cabecalhos.get("content-type") ?? "";
exigir(tipo.includes("application/problem+json"), `esperava application/problem+json, veio ${tipo}`);
for (const campo of ["type", "title", "status", "detail", "instance", "erro", "mensagem", "req_id"]) {
  exigir(campo in objeto, `Problem Details sem o campo ${campo}: ${Object.keys(objeto).sort().join(",")}`);
}
exigir(objeto.status === 401, `campo status divergente do HTTP: ${objeto.status} != 401`);
console.log(`${objeto.type} -> ${objeto.detail} (req_id ${objeto.req_id})`);

File: exemplos/python/10_erro_problem_details.py
# Formato de erro da API: RFC 9457 (application/problem+json) com os campos da casa junto
from _apoio import chamar, exigir

status, cabecalhos, corpo = chamar("/api/eu", chave="plat_" + "z" * 43)
exigir(status == 401, f"esperava 401 com chave inválida, veio {status}: {corpo}")
tipo = cabecalhos.get("Content-Type", cabecalhos.get("content-type", ""))
exigir("application/problem+json" in tipo, f"esperava application/problem+json, veio {tipo!r}")
for campo in ("type", "title", "status", "detail", "instance", "erro", "mensagem", "req_id"):
    exigir(campo in corpo, f"Problem Details sem o campo {campo}: {sorted(corpo)}")
exigir(corpo["status"] == 401, f"campo status divergente do HTTP: {corpo['status']} != 401")
print(f"{corpo['type']} -> {corpo['detail']} (req_id {corpo['req_id']})")
